protocol/hello/hello.go

```diff
@@ -176,7 +198,7 @@ func (hn *helloNotify) Connected(n net.Network, c net.Conn) {
 		ctx, cancel := context.WithTimeout(context.Background(), helloTimeout)
 		defer cancel()
 		p := c.RemotePeer()
-		if err := hn.hello().sayHello(ctx, p); err != nil {
+		if _, err := hn.hello().SayHello(ctx, p); err != nil {
```

Comment:
Sending a hello message on this call and then again in the handler is not optimal, but it's easy, doesn't require changing the protocol bits, and doesn't rewrite much of the hello code. If this extra sending is not acceptable a proposed alternative could be to:
- modify the method `handleNewStream` such that it only sends a hello message.
- modify `SayHello` to read a message from a steam
- modify `Connect` to send a hello message, receive a response, process it and trigger the callback.

Reply:
I'm fine with this for now.  Please put up a TODO with a follow on issue for investigating making this simpler.

Reply:
implemented here: https://github.com/filecoin-project/go-filecoin/pull/3327/commits/06dbf50930399ea4c028944adf47e9c5fe8a1776, turned out to be cleaner than anticipated.